src/base/castlerandom.pas

```diff
@@ -67,22 +68,50 @@ procedure TCastleRandom.Initialize(RandomSeed: LongWord);
 begin
   if RandomSeed = 0 then
   begin
+    Seed := GetRandomSeed;
+  end
+  else seed := LongInt(RandomSeed);
+end;
+
+{$IFDEF Linux}
+function DEV_URANDOM: longint;
+var dev_rnd: file of integer;
+begin
+  { algorithm according to http://wiki.freepascal.org/Dev_random
+   /dev/urandom is a native *nix very high-quality random number generator.
+   it's 1000 times slower than CastleRandom,
+   but provides a perfect seed initialization. }
+  //filemode := 0;
+  AssignFile(dev_rnd, '/dev/urandom');
+  reset(dev_rnd);
+  read(dev_rnd,result);
+  CloseFile(dev_rnd);
+end;
+{$ENDIF}
+
```

Comment:
I've made a new random initialization algorithm yesterday evening. There are several problems I was solving of SysUtils.Randomize. It's horribly not-thread safe and our current implementation also wasn't (today's morning tests showed that initializing 16 randoms in 16 threads gives equal seeds for some of them). I hope I've solved this problem (by not a very safe xor-ring QWORD(@c64) which is unique for each thread even if gettickcount64 returns equal results, but I need to test it thoroughly.